src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableEditorManager.class.php

```diff
@@ -502,6 +496,16 @@ public function RemoveMLTConnection($sFieldName, $iConnectedID)
         }
     }
 
+    private function addInverseRestrictingMLTConnection(string $restrictionField, string $restriction): void
+    {
+        $sourceTable = substr($restrictionField, 0, -4);
```

Comment:
MLT connections can also be configured (?) then the name is arbitrary and doesn't have to end with _mlt.

There should be some method somewhere there gives the (target) table name for a field.

Reply:
I copied the code over from where it was before - but you are right, this cannot work with customized MLT Table names

Reply:
Surprisingly, this part of the code works fine: The iframe for the relation is the table editor with a restriction field added to it. This restriction field does not actually exist on the target table but correctly resolves to the source table's table editor which is then used to add the relation. However, this part a bit further down does not work as expected:

```php
$tableEditor = new TCMSTableEditorManager();
$tableEditor->Init(TTools::GetCMSTableId($sourceTable), $restriction);
$tableEditor->AddMLTConnection($targetTable . '_mlt', $this->sId);
```

Specifically the last line expects the column to adhere to the `{TARGET_TABLE}_mlt` naming convention.

Reply:
:heavy_check_mark: Fixed by also passing through the `field` property containing the origin field name and using that in the table editor. Can you carefully review https://github.com/chameleon-system/chameleon-base/pull/528/commits/c25392998cb0a4db6f7ef6cf3313275b3e2982bb in particular since I am not 100% sure which unintended consequences the change to `TCMSListManagerEndPoint`  could have?